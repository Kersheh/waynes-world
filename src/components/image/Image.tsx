import React, { useState, useEffect } from 'react';
import classNames from 'classnames/bind';

import styles from './Image.module.scss';

const cx = classNames.bind(styles);

interface ImageProps {
  src: string;
  alt?: string;
  className?: any;
}
const Image = ({ src, alt, className }: ImageProps) => {
  const [isLoaded, setIsLoaded] = useState(false);

  useEffect(() => {
    setIsLoaded(false);
  }, [src]);

  return (
    <div className={styles.image}>
      <img
        style={isLoaded ? {} : { display: 'none' }}
        src={src}
        alt={alt}
        onLoad={() => setIsLoaded(true)}
        className={className}
      />
      {!isLoaded && (
        <div
          className={cx({
            [className]: true,
            placeholderWrapper: true
          })}
        >
          <img
            src="/images/placeholder.svg"
            alt="Loading placeholder"
            className={styles.placeholder}
          />
        </div>
      )}
    </div>
  );
};

export default Image;
