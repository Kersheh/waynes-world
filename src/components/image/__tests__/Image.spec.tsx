import React from 'react';
import { render } from '@testing-library/react';

import Image from '../Image';

describe('<Image/> component', () => {
  it('should render', () => {
    const { container } = render(<Image src="/some/image/src" />);

    expect(container).toMatchSnapshot();
  });
});
