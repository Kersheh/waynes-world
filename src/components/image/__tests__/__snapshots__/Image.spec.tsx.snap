// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> component should render 1`] = `
<div>
  <div
    class="image"
  >
    <img
      draggable="false"
      src="/some/image/src"
      style="display: none;"
    />
    <div
      class="undefined placeholderWrapper"
    >
      <img
        alt="Loading placeholder"
        class="placeholder"
        src="/images/placeholder.svg"
      />
    </div>
  </div>
</div>
`;
